src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -47,8 +48,10 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
             {
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
+                // in case of SingleFile deployment, Assembly.Location is empty.
+                isSingleFile = exeAssembly != null && string.IsNullOrEmpty(exeAssembly.Location);
 
-                if (exeAssembly != null)
+                if (exeAssembly != null && !isSingleFile)
                 {
                     HasEntryAssembly = true;
```

Comment:
`HasEntryAssembly` should still be set to true, even when the entry assembly is loaded from memory.